caikit_nlp/modules/token_classification/transformer_sentence_classification.py

```diff
@@ -0,0 +1,218 @@
+        )
+        error.type_check(
+            "<NLP56932573E>", Dict, allow_none=True, labels_mapping=labels_mapping
+        )
+        self.lang = lang
+        self.sentence_splitter = sentence_splitter
+        self.sequence_classifier = sequence_classifier
+        self.default_threshold = default_threshold
+        self.labels_to_output = labels_to_output
+        self.labels_mapping = labels_mapping
+
+    ################################## API functions #############################################
+
+    def run(
+        self, text: str, threshold: Optional[float] = None
+    ) -> TokenClassificationResult:
+        """Run sentence-level classification on text. Returns results
+        based on score threshold for labels that are to be outputted
+
+        Args:
+            text: str
+                Document to run sentence-level classification on
+            threshold: float
+                (Optional) Threshold based on which to return score results
+
+        Returns:
+            TokenClassificationResult
+        """
+        if threshold is None:
+            threshold = self.default_threshold
+        token_classification_results = []
+        # Split document into sentences
+        sentence_span_list = self.sentence_splitter.run(text)
+        # Run each sentence through the classifier and determine based
+        # on threshold and labels_to_output what results should be returned
+        text_list = [span.text for span in sentence_span_list]
+        classification_results = self.sequence_classifier.run_batch(text_list)
+        for idx, classification_result in enumerate(classification_results):
+            # Each classification prediction is list of classifications
+            # for that particular text example
+            for classification in classification_result.results:
+                # Map labels to semantic labels if provided
+
+                # NOTE: labels need to be specified as str for config
+                label = str(classification.label)
+                if self.labels_mapping:
+                    # Use original classifier label if not found
+                    label = self.labels_mapping.get(label, label)
+                if classification.score >= threshold:
+                    if not self.labels_to_output or (
+                        self.labels_to_output and label in self.labels_to_output
+                    ):
+                        sentence_span = sentence_span_list[idx]
+                        token_classification = TokenClassification(
+                            start=sentence_span.start,
+                            end=sentence_span.end,
+                            word=sentence_span.text,
+                            entity=label,
+                            score=classification.score,
+                        )
+                        token_classification_results.append(token_classification)
```

Comment:
Can we let this filtering to the application / user? That would be more transparent I feel and provide them opportunity to make the decision based on their particular need and model.

Reply:
The use case we are trying to support with this asked for thresholding. I have tried to put some of the "decision making" with the particular model via `default_threshold` stored with the model. I do agree this is definitely one of the cases where an application/user _could_ do the work.